tests/Microsoft.VisualStudio.ProjectSystem.Managed.VS.UnitTests/ProjectSystem/VS/Properties/VBDefineConstantsEncodingTests.cs

```diff
@@ -0,0 +1,135 @@
+        [InlineData("key1=/\"value1/\",key2=/\"value2/\",key3=/\"value3/\"", new[] { "key1", "\"value1\"", "key2", "\"value2\"", "key3", "\"value3\"" })]
+        [InlineData("something=/\"equals/=this/\"", new[] { "something", "\"equals=this\"" })]
+        [InlineData("oh=/\"hey/=there/\",hi=/\"didnt/=see/\",you=/\"there/\"", new[] { "oh", "\"hey=there\"", "hi", "\"didnt=see\"", "you", "\"there\"" })]
+        [InlineData("path=/\"//path//to//somewhere/\"", new[] { "path", "\"/path/to/somewhere\"" })]
+        [InlineData("file=/\"//path//is//here/\"", new[] { "file", "\"/path/is/here\"" })]
+        [InlineData("files=/\"path/=//path//to//somewhere/\"", new[] { "files", "\"path=/path/to/somewhere\"" })]
+        [InlineData("a=1,a=2", new[] { "a", "1", "a", "2" })]
+        [InlineData("key1  =/\"value1  /\"", new[] { "key1  ", "\"value1  \"" })]
+        [InlineData("  key1=/\"  value1/\"", new[] { "  key1", "\"  value1\"" })]
+        [InlineData("key1=/\"a b c d/\"", new[] { "key1", "\"a b c d\"" })]
+        [InlineData("key1=/\"a/=b/=c/=d/\"", new[] { "key1", "\"a=b=c=d\"" })]
+        [InlineData("key1=/\"a/=b/\"/\"c/=d/\"", new[] { "key1", "\"a=b\"\"c=d\"" })]
+        [InlineData("key1=/\"/\"ab/\"/\"cd/\"", new[] { "key1", "\"\"ab\"\"cd\"" })]
+        [InlineData("key1=/\"/\"/\"ab/\"/\"cd/\"/\"ef/\"", new[] { "key1", "\"\"\"ab\"\"cd\"\"ef\"" })]
+        [InlineData("key1=/\"/\"/\"ab/\"/\"cd/\"", new[] { "key1", "\"\"\"ab\"\"cd\"" })]
+        [InlineData("a b c d=/\"value1/\"", new[] { "a b c d", "\"value1\"" })]
+        [InlineData("a=/\"b/=c/=d/=value1/\"", new[] { "a", "\"b=c=d=value1\"" })]
+        [InlineData("a=/\"b/\"/\"c/=d/=value1/\"", new[] { "a", "\"b\"\"c=d=value1\"" })]
+        [InlineData("/\"ab/\"/\"cd=/\"value1/\"", new[] { "\"ab\"\"cd", "\"value1\"" })]
+        [InlineData("/\"/\"ab/\"/\"cd/\"/\"ef=/\"value1/\"", new[] { "\"\"ab\"\"cd\"\"ef", "\"value1\"" })]
+        [InlineData("/\"/\"ab/\"/\"cd=/\"value1/\"", new[] { "\"\"ab\"\"cd", "\"value1\"" })]
+        [InlineData("a//bc=/\"value1/\"", new[] { "a/bc", "\"value1\"" })]
+        [InlineData("", new string[0])]
+
+        [InlineData("a=1,b=2", new[] { "a", "1", "b", "2" })]
+        [InlineData("a=1123,b=456", new[] { "a", "1123", "b", "456" })]
+        [InlineData("a=1,b=/\"apple/\"", new[] { "a", "1", "b", "\"apple\"" })]
+        [InlineData("a=1,b=/\"apple/\",c=456", new[] { "a", "1", "b", "\"apple\"", "c", "456" })]
+        [InlineData("a=1,b=/\"apple/\",c=456,d=True", new[] { "a", "1", "b", "\"apple\"", "c", "456", "d", "True" })]
+
+        [InlineData("a=True", new[] { "a", "True"})]
+        [InlineData("a=False", new[] { "a", "False" })]
+        [InlineData("a=True,b=False", new[] { "a", "True", "b", "False" })]
+        [InlineData("a=True,b=False,c=123", new[] { "a", "True", "b", "False", "c", "123" })]
+        public void NameValuePairsVBDefineConstantsEncoding(string encodedPairs, string[] pairs)
```

Comment:
I'm trying to think of a good name for this method, maybe `ValidateCorrectPairsAreGenerated`?